hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -203,6 +208,45 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Opti
     return true;
   }
 
+  public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Option<Map<String, String>> extraMetadata,
+                             String commitActionType, Map<String, List<String>> partitionToReplaceFileIds) {
+
+    HoodieCommitMetadata metadata = CommitUtils.buildMetadata(stats, partitionToReplaceFileIds, extraMetadata,
+        operationType, config.getSchema(), commitActionType);
+    if (config.isMultiWriterEnabled()) {
+      // get strategy and lock type
+      LockConfiguration lockConfiguration = new LockConfiguration(config.getProps());
+      LockProvider lockProvider = (LockProvider) ReflectionUtils.loadClass(config.getLockProviderClass(),
```

Comment:
can we move the reflection part to the config object?

Reply:
Already have it that way for `ConflictResolutionStrategy`, but this reflection requires parameters hence kept it here. If we moved this to config object, we have to make the getLockProviderClass parameterized which will be a problem for clients to get just the class name. Other option is to have 2 overloaded getters. Let me know what you think.